pkg/api/types.go

```diff
@@ -1205,6 +1221,13 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// Local represents a directory with specific topology contraints
+type LocalVolumeSource struct {
+	// The full path to the volume on the node
+	// For alpha, this path must be a file-based volume
+	Path string
```

Comment:
Thanks Saad for the detailed explanation and highlighting the pros and cons of each approach.  I think we won't be able to come up with a scheme that perfectly addresses each concern, so I am leaning towards option 2, or having two explicit fields, PathToDevice, or PathToMountedDirectory.  @jsafrane what do you think?